Custom Notification Dot Counter Color Support

yasanglass opened the pull request:
## Description

Adds custom color support to Notification Dots' counter text.

Requires [platform_frameworks_libs_systemui#6](https://github.com/LawnchairLauncher/platform_frameworks_libs_systemui/pull/6).

Fixes #2923

## Type of change
<!-- Replace :x: with :white_check_mark: to "check" the specified bullet -->

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

yasanglass committed:
Add `notificationDotTextColor` to `PreferenceManager2`

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Add `notificationDotTextColor` to preferences UI

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Pass counter color to `DotRenderer`

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Convert `LauncherDefault` to `Default`

- Renames LauncherDefault class to Default
- Change default `launcher_default` & `app_icon` configs to `default`

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Optimize imports

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Rename "Default" to "Managed by Lawnchair"

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Use the correct `ColorDot` on `ColorPreference`

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Show a contrast warning for dot colors

Shows a warning when notification dot & dot counter colors do not have enough contrast with each other

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasandev commented:
5ac959a71e50a6feff03a25438c6888791f30f02 Adds a warning that is shown whenever the dot background & foregound (counter text) colors do not have enough contrast with each other.

[ColorContrastWarning](https://github.com/LawnchairLauncher/lawnchair/pull/2977/commits/5ac959a71e50a6feff03a25438c6888791f30f02#diff-896a87a728219ae3a62e4b9efa7f2b25894b68445ec66b5d45e0fe68402bddfd) composable can be used for any other two colors when needed later on.

![Screenshot_20220926_122902](https://user-images.githubusercontent.com/41836211/192236277-bedb6afb-49c5-480a-8164-5f6e83f3608b.png)

yasanglass committed:
Show a special icon on `ColorDot` when the color is 0

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasandev commented:
647f8efcbfed4e7f3aa32ba7799c9705cb535dde changes `ColorDot` so it shows an "A" icon when the color is _**Managed by Lawnchair**_ (color is 0). 

![Screenshot from 2022-09-26 17-15-53](https://user-images.githubusercontent.com/41836211/192292955-b0fb48bd-4b58-4ea7-86b1-4222ba911ecf.png)![Screenshot from 2022-09-26 17-15-44](https://user-images.githubusercontent.com/41836211/192292964-cab3cca2-da2d-4cf6-9929-a7819162a5a4.png)

yasanglass committed:
Make `ColorDot(Color)` private to avoid accidental usages

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Create `WarningPreference` as a reusable composable for warnings

Unlike `ColorContrastWarning`, `WarningPreference` only requires a warning text and can be used for anything.

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Show contrast warning if dot color is dynamic and the dot text color is not `default`

The `default` color for dot & dot text colors behave differently:
   - default dot color -> picked automatically based on icon color
   - default dot text color -> picked automatically to have enough contrast with the dot color

As a result, if the dot color is dynamic but the text color is not `default`, the colors can have not enough contrast with each other in on some icons.

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasandev commented:
d84b8ea441943eba477ae03f50c485048ae4a440 adds a new custom scenario for when the contrast warning is shown:

The `default` color for dot & dot text colors behave differently:
   - default dot color -> picked automatically based on icon color
   - default dot text color -> picked automatically to have enough contrast with the dot color

As a result, if the dot color is dynamic (changes based on the app icon, system theme, or wallpaper) but the text color is not `default` (picked automatically based on the dot color), the colors can sometimes **not** have enough contrast with each other in some scenarios.

![Screenshot from 2022-09-26 17-43-38](https://user-images.githubusercontent.com/41836211/192305145-41904dbd-f608-41f8-bd96-34f540a7a363.png)

yasanglass committed:
Merge branch '12.1-dev' into feature/custom-notification-dot-text-color

suphon-t commented on lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorContrast Warning.kt:
```diff
@@ -0,0 +1,63 @@
+package app.lawnchair.ui.preferences.components.colorpreference
```
Looks like there's a space in the file name

yasanglass replied:
That's weird. Fixed it now.

suphon-t commented on lawnchair/src/app/lawnchair/ui/preferences/components/WarningPreference.kt:
```diff
@@ -0,0 +1,33 @@
+        },
+        startWidget = {
+            Icon(
+                imageVector = Icons.Outlined.Warning,
```
Please change it to use the rounded icons to be consistent with other part of the UI.

yasanglass replied:
My bad, fixed.

yasanglass committed:
Use Rounded icon on `WarningPreference`

Co-authored-by: Daria Hamrah Paytakht <info@dariarnd.ir>

yasanglass committed:
Fix `ColorContrastWarning.kt`'s file name

Goooler commented on lawnchair/src/app/lawnchair/ui/preferences/components/colorpreference/ColorDot.kt:
```diff
@@ -17,27 +21,53 @@ fun <T> ColorDot(
+    val context = LocalContext.current
+
+    val colorLight = entry.lightColor(context)
+    val colorDark = entry.darkColor(context)
```
Unnecessary change?

yasanglass replied:
I created the variables because I needed to use `colorLight` on line 35.

yasanglass committed:
Update platform_frameworks_libs_systemui
